ext/fileinfo/libmagic/magic.c

```diff
@@ -403,3 +403,12 @@ magic_getparam(struct magic_set *ms, int param, void *val)
 		return -1;
 	}
 }
+
+public const char *
+magic_magic_file_get(struct magic_set *ms)
+{
+	if(strncmp(ms->file, "unknown", strlen("unknown")) == 0) {
```

Comment:
`magic_map->p == php_magic_database` were to know. Perhaps one could return "internal" as a name. Alternatively yes, if you want to base on the filename presence, I would suggest to return an explicit `NULL` for the internal db.